lightningd/plugin.c

```diff
@@ -1273,6 +1273,44 @@ static void plugin_manifest_timeout(struct plugin *plugin)
 		fatal("Can't recover from plugin failure, terminating.");
 }
 
+static const char *plugin_notifications_add(const char *buffer,
+					    const jsmntok_t *result,
+					    struct plugin *plugin)
+{
+	char *name;
+	const jsmntok_t *method, *obj;
+	const jsmntok_t *notifications =
+	    json_get_member(buffer, result, "notifications");
+
+	if (!notifications)
+		return NULL;
+
+	if (notifications->type != JSMN_ARRAY)
+		return tal_fmt(plugin,
+			       "\"result.notifications\" is not an array");
+
+	for (size_t i = 0; i < notifications->size; i++) {
+		obj = json_get_arr(notifications, i);
```

Comment:
`json_for_each_arr(i, obj, notifications)` does these two lines, BTW (and slightly more efficiently).